File: chapter05/akkademaid-scala-5/src/main/scala/com/lp/akka/notes/ArticleParser.scala
package com.lp.akka.notes

/**
 * @author li.pan
 * @version 1.0.0
 * @Description TODO
 * @createTime 2021年01月14日 13:03:00
 */
object ArticleParser {
  def apply(html: String): String = de.l3s.boilerpipe.extractors.ArticleExtractor.INSTANCE.getText(html)
}

File: chapter05/akkademaid-java-5/src/main/java/com/lp/akka/notes/ArticleParser.java
package com.lp.akka.notes;

import com.jasongoodwin.monads.Try;

/**
 * @author li.pan
 * @version 1.0.0
 * @Description 用 BoilerPipe 库来解析页面。
 * @createTime 2021年01月14日 12:47:00
 */
public class ArticleParser {
    public static Try<String> apply(String html) {
        return Try.ofFailable(
                () -> de.l3s.boilerpipe.extractors.ArticleExtractor.INSTANCE.getText(html)
        );
    }
}
